### ISSUE TITLE
Nested click2load frames not working

### ISSUE THREAD

**Issue by umbralOptimatum** [uBlockOrigin/uBlock-issues#3088]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

I'm trying to use these rules:

```
||youtube.com^$3p,frame,redirect=click2load.html
||twitch.tv^$3p,frame,redirect=click2load.html
||safe.txmblr.com^$3p,frame,redirect=click2load.html
```

Tumblr wraps embed iframes in a `safe.txmblr.com` iframe. When a page has embedded Youtube or Twitch videos, clicking the click2load for `safe.txmblr.com` loads a second click2load for the embedded video, but clicking it does not load the video or have any other visible effect.

### A specific URL where the issue occurs.

```
https://thefloatingstone.tumblr.com/post/727730225735204864
```

### Steps to Reproduce

1. Load the above Tumblr page with the two rules enabled.
2. Click the click2load embed and a second click2load embed should appear.
3. Click the second click2load embed and nothing visibly happens.

### Expected behavior

Click2load should work even when loaded by another click2load.

### Actual behavior

Original embed never loads.

### uBO version

1.55.0

### Browser name and version

Firefox 115.6.0esr

### Operating System and version

macOS 10.14.6

**Comment by garry-ut99** [comment]
On Opera/Chromium there is no issue, it always works as expected : 
`the first click2load iframe appears` ==> `first click` ==> 
`then second click2load iframe apepars` ==> `second click` ==> `video loads fine`

On Firefox (115 ESR) the behaviour is unexpected/random :
- sometimes video works just fine like above on Opera (after 2 clicks)
- sometimes the video loads prematurely after the 1st click (no need for the 2nd click then)
- sometimes the second click doesn't work for the 1st time, but works for the 2nd time or 3rd, 4th etc.
- sometimes the second click never works, no matter how many clicks

**Comment by gorhill** [comment]
I spent some time yesterday investigating this.

First, it works reliably in CHromium-based browsers.

Now in Firefox I cannot explain the behavior. I added multiple `console.log` at key points, and I definitely do see the `self.location.replace()` call being made with the Youtube URL, but it has no effect -- **until** some time has passed. Sometimes it works immediately, often there needs to be a delay of up to 30s before it works, but it eventually always work, just with some delay. I have also seen the behavior that allowing Tumblr iframe immediately also allows the YT iframe, so it's extremely inconsistent in Firefox and so far I cannot explain it other than by Firefox doing unexpected things internally.

**Comment by gorhill** [comment]
> clicking a few extra times loaded it instantly

Yes, it's what I meant, if we keep clicking it works eventually, sometimes after 10x times. Something inside Firefox is preventing the navigation from taking place. I didn't try to synthesize a minimal case.

**Comment by gwarser** [comment]
This works fine:

http://0.0.0.0:8080/iframe-root.html

`iframe-root.html`
```html
<!DOCTYPE html>
<html>
        <head> 
        </head>
        <body style="background: yellow;">
        <div><iframe src="http://0.0.0.0:8080/iframe.html" style="width: 50%; height: 600px; background: green;"></iframe>
        </body>
</html>
```

`iframe.html`
```html
<!DOCTYPE html>
<html>
        <head> 
        </head>
        <body>
        <div><iframe src="https://example.com/" style="width: 100%; height: 300px; background: blue;"></iframe>
        </body>
</html>
```

```abp
||example.com^$3p,frame,redirect=click2load.html
||0.0.0.0^$frame,redirect=click2load.html
```

**Comment by garry-ut99** [comment]
I'm not sure whether it will be any  useful or not, anyway, short description: 
while clicking the iframe, the following things can be observed:
- the browser refresh button is animatiing
- and the iframe document DOM content is being highlighted in yellow which means it keeps being refreshed but the content of it is the same all the time (I verified it by expanding it each time it gets highlighted in yellow)

[ANIMATION](https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/0ad33855-0804-4bab-afa7-3dabe9df4d77)
